### DISCUSSION TITLE
Why the name of the function is called client.oauthCallback while we are exchanging the token

### DISCUSSION THREAD

**Discussion by rahimimo** [panva/openid-client#548] [Q&A]
I am new to oAuth, however I saw the draft of open-id and I did not understand where does the name of client.oauthCallback() is coming from. The name was a bit misleading at least for me. Should not be getToken or exchangeCodeWithToken for instance?
This function is not passed to us so to my understanding it is not a callback.

**Comment by rahimimo** [comment]
I spent some more time and I think there are two meaning of the term "callback"
1. Some function passed as arg https://www.codefellows.org/blog/what-is-a-callback-anyway/#:~:text=In%20computer%20programming%2C%20a%20callback,as%20in%20an%20asynchronous%20callback.
2. Meanaing postback https://www.adjust.com/glossary/callback/

Does this callback closer to postback? I traiced back the callback and it is actually a function which is not passed in the earlier step

**Comment by rahimimo** [comment]
More to the definition of the term callback from MDN which is not aligned with this use case (client.oauthCallback) https://developer.mozilla.org/en-US/docs/Glossary/Callback_function

**Comment by Tyrannogyna** [comment]
Agree, this is very confusing. As a little extra, I also don't understand what URL this callback method is expecting, the docs are not clear.
`redirect_uri used for the authorization request` is this the enpoint we used in the request to get the auth code, the endpoint of the auth server to retrieve the token, the uri we expect the token to be returned to, something else?
